docs/v3/core/v3.0.rst

```diff
@@ -1211,40 +1210,42 @@ the following procedure:
 -----------------
 
-To allow for flexibility to define and implement new codecs, this
-specification does not define any codecs, nor restrict the set of
-codecs that may be used. Each codec must be defined via a separate
-specification. In order to refer to codecs in array metadata
-documents, each codec must have a unique identifier, which is a URI
-that dereferences to a human-readable specification of the codec. A
-codec specification must declare the codec identifier, and describe
-(or cite documents that describe) the encoding and decoding algorithms
-and the format of the encoded data.
-
-A codec may have configuration parameters which modify the behaviour
-of the codec in some way. For example, a compression codec may have a
-compression level parameter, which is an integer that affects the
-resulting compression ratio of the data. Configuration parameters must
-be declared in the codec specification, including a definition of how
-configuration parameters are represented as JSON.
-
-The Zarr core development team maintains a repository of codec
-specifications, which are hosted alongside this specification in the
-`zarr-specs GitHub repository`_, and which are
-published on the `zarr-specs documentation Web site
-<https://zarr-specs.readthedocs.io/>`_. For ease of discovery, it is
-recommended that codec specifications are contributed to the
-zarr-specs GitHub repository. However, codec specifications may be
-maintained by any group or organisation and published in any location
-on the Web. For further details of the process for contributing a
-codec specification to the zarr-specs GitHub repository, see
-`ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ which describes
-the process for Zarr specification changes.
-
-Further details of how codecs are configured for an array are given in the `Array metadata`_ section.
+.. _core-codecs:
+
+Core codecs
+-----------
+
+This spec defines a set of codecs ("core codecs") which all Zarr implementations SHOULD implement in 
+order to ensure a minimal level of interoperability between Zarr implementations.
+Each core codec is each defined by specification documents which are hosted in the
+`zarr-specs GitHub repository`_, and are published on the `zarr-specs documentation web site
+<https://zarr-specs.readthedocs.io/>`_. The list of core codecs is part of the Zarr v3 specification. 
+Changes to the list of core codecs MUST be made via the same protocol used for 
+changing the Zarr v3 specification. Changes to the list of core codecs SHOULD be made carefully and
+in close collaboration with extant Zarr v3 implementations. A new core codec SHOULD be added to the 
+list when a sufficient number of Zarr implementations support or intend to support that codec. 
+An existing core codec SHOULD be removed from the list when a sufficient number of implementation 
+developers and Zarr users deem the codec worth removing, e.g. because of a technical flaw in the 
+algorithm underlying the codec.
```

Comment:
One of the main goals (to me) of specifying core codecs is to ensure backwards compatibility long into the future. Removing a codec reduces backwards compatibility, so I think there should be very strict conditions under which a codec should be removed.

```suggestion
Existing core codecs SHOULD NOT be removed from the list to ensure future backwards compatibiility.
A core codec MAY be removed from the list if it has fundamental security flaws that cannot be resolved and the removal is agreed by the Zarr Implementation Council.
```

Reply:
I don't think we need a reference to the ZIC here, since elsewhere in the PR I state that the list of core codecs should be changed according to the same process for changing the spec itself

Reply:
```suggestion
Existing core codecs SHOULD NOT be removed from the list to ensure future backwards compatibiility.
A core codec MAY be removed from the list if it has fundamental security flaws that cannot be resolved.
```

Reply:
"To ensure backwards compatibility, existing core codecs SHOULD NOT be removed from the list." reads better to me

Reply:
also something I thought of last night: if a codec is removed from the list of core codecs, it is still available as an extension codec. So I don't think there's actually a large backwards compatibility risk.